### PR TITLE
Ins 1038 inspect app users

### PR DESCRIPTION
**Pull request by fwpushan**

### CODE REVIEW

**Review by andrea-williams — APPROVED** [review]
✅ Jenkins build passed
✅ SonarQube passed
✅ manual check in DEV env - everything looks good
✅ skimmed through the code manually - no issues found but there's too many code changes to check it thoroughly

File: api/api_sources/sources/server/core/base.route.controller.ts [hunk 1]
```diff
@@ -203,12 +206,13 @@ export class RouteController {
...
-
+        // console.log(`${this.className}`);
         // Apply config to route
-        _.each(this.configs, (config: RouteConfig) => {
+        for (const config of this.configs) {
             try {
                 // Getting endpoint
-                let endPoint: string = config.description.path.split('#')[1];
+                const paths: string[] = config.description.path.split('#');
+                let endPoint: string = paths.length > 0 ? paths[1] : paths[0];
```
Comment (AmirShayegh) [inline api/api_sources/sources/server/core/base.route.controller.ts]: 👍

File: app/lucy/src/app/components/Routes/admin-tools/admin-tools.component.ts [hunk 2]
```diff
@@ -50,19 +52,43 @@ export class AdminToolsComponent implements OnInit, AfterViewInit {
...
+
   requestLength(): number {
     return this.requests.length;
   }  
 
   constructor(
-    private roles: RolesService,
-    private admin: AdminService,
+    private roleService: RolesService,
+    private adminService: AdminService,
```
Comment (AmirShayegh) [inline app/lucy/src/app/components/Routes/admin-tools/admin-tools.component.ts]: 👍 nice refactoring

File: app/lucy/src/app/components/Routes/admin-tools/request-modal/request-modal.component.ts [hunk 3]
```diff
@@ -149,10 +162,11 @@ export class RequestModalComponent implements OnInit {
     const selected = this.roles.find(item => item.role === this.selectedRole);
     try {
       if (requester.accountStatus === 0) {
-        await this.adminService.changeUserAccountStatus(requester, requester.accountStatus);
+        await this.adminService.changeUserAccountStatus(requester, requester.accountStatus, this.isAdmin);
       }
       this.request.requester.roles = [selected];
-      await this.adminService.respondToRequest(this.request, true);
+      const status = await this.adminService.respondToRequest(this.request, true);
+      if (!status) throw new Error();
```
Comment (AmirShayegh) [inline app/lucy/src/app/components/Routes/admin-tools/request-modal/request-modal.component.ts]: 👍

**Review by AmirShayegh — APPROVED** [review]

**Pull request merged**